overlord/snapstate/handlers.go

```diff
@@ -1280,6 +1275,56 @@ func daemonRestartReason(st *state.State, typ snap.Type) string {
 	return "Requested daemon restart (snapd snap)."
 }
 
+// maybeUndoRemodelBootChanges will check if an undo needs to update the
+// bootloader. This can happen if e.g. a new kernel gets installed. This
+// will switch the bootloader to the new kernel but if the change is later
+// undone we need to switch back to the kernel of the old model.
+func maybeUndoRemodelBootChanges(t *state.Task, undoInfo *snap.Info) error {
+	// get the new and the old model
+	deviceCtx, err := DeviceCtx(t.State(), t, nil)
+	if err != nil {
+		return err
+	}
+	// we only have an old model if we are in a remodel situation
+	oldModel := deviceCtx.OldModel()
+	if oldModel == nil {
+		return nil
+	}
+	newModel := deviceCtx.Model()
+	if undoInfo.InstanceName() != newModel.Kernel() {
```

Comment:
this is weird  we check kernel only here, but below there is code also for bases?  do we need a TODO here that says base support will come?